homeassistant/components/dyson/air_quality.py

```diff
@@ -1,55 +1,43 @@
 """
 Support for Dyson Pure Cool Air Quality Sensors.
-
-For more details about this platform, please refer to the documentation at
-https://home-assistant.io/components/dyson/
 """
 import logging
 
-from homeassistant.components.air_quality import (
-    AirQualityEntity, ATTR_ATTRIBUTION, ATTR_NO2, ATTR_PM_10, ATTR_PM_2_5)
-from homeassistant.components.air_quality import DOMAIN as AIQ_DOMAIN
-from homeassistant.components.dyson import DYSON_DEVICES, DOMAIN
+from homeassistant.components.air_quality import AirQualityEntity
+from . import DYSON_DEVICES
 
 DEPENDENCIES = ['dyson']
 
 ATTRIBUTION = 'Dyson purifier air quality sensor'
 
 _LOGGER = logging.getLogger(__name__)
 
+DYSON_AIQ_DEVICES = 'dyson_aiq_devices'
+
 ATTR_VOC = 'volatile_organic_compounds'
 
-PROP_TO_ATTR = {
-    'attribution': ATTR_ATTRIBUTION,
-    'nitrogen_dioxide': ATTR_NO2,
-    'particulate_matter_10': ATTR_PM_10,
-    'particulate_matter_2_5': ATTR_PM_2_5,
-    'volatile_organic_compounds': ATTR_VOC,
-}
+AIQ_SUFFIX = "-AIQ"
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Dyson Sensors."""
     from libpurecool.dyson_pure_cool import DysonPureCool
 
-    _LOGGER.debug("Creating new Dyson air quality sensors")
-    device_serials = []
-    devices = []
-
     if discovery_info is None:
         return
 
-    if AIQ_DOMAIN in hass.data:
-        device_serials = {entity.device_serial for entity
-                          in hass.data[AIQ_DOMAIN].entities
-                          if entity.platform.platform_name == DOMAIN}
+    _LOGGER.debug("Creating new Dyson air quality sensors")
+
+    if DYSON_AIQ_DEVICES not in hass.data:
+        hass.data[DYSON_AIQ_DEVICES] = []
 
     # Get Dyson Devices from parent component
+    device_ids = [device.unique_id for device in hass.data[DYSON_AIQ_DEVICES]]
     for device in hass.data[DYSON_DEVICES]:
         if isinstance(device, DysonPureCool) and \
-                device.serial not in device_serials:
-            devices.append(DysonAirSensor(device))
-    add_entities(devices)
+                device.serial + AIQ_SUFFIX not in device_ids:
```

Comment:
Use new style string formatting instead of string concatenation.

Reply:
Done.